### PR TITLE
fix: remove updateTransactionRequestHook from allowance read call

### PR DESCRIPTION
**Pull request by Abhikumar98**
https://lifi.atlassian.net/browse/LF-4515

### CODE REVIEW

File: src/allowance/checkAllowance.unit.spec.ts
Comment (Addminus) [inline src/allowance/checkAllowance.unit.spec.ts:1]: Why did you delete the unit test?

Comment (Abhikumar98) [reply src/allowance/checkAllowance.unit.spec.ts:1]: updated and brought back the regular allowance call

**Review by Addminus — APPROVED** [review]

**Review by chybisov — APPROVED** [review]
@Abhikumar98 please use meaningful messages as PR titles because these titles then become CHANGELOG records 🙂

Also, https://www.conventionalcommits.org/

@Addminus I noticed this issue while migrating to viem, which is already fixed in the viem PR. I thought it didn't break anything in the existing release, but it appears it does 🙂

Also, we need to rewrite the test. It doesn't do much now, but probably leave this to after the viem era 🙂
